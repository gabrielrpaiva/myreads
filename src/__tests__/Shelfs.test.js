import React from 'react';
import { shallow, mount } from "enzyme";
import ShelfsBooks from '../Shelfs';


const allShelfs = {
    moveto: "Move to..",
    currentlyReading: "Currently Reading",
    wantToRead: "Want To Read",
    read: "Read",
    none: "None"
}


const setUpdate = jest.fn()

const bookIdUpdate = ""

const filterRelatedBooks = jest.fn()

const classPopUp = ""

const relatedBooks = []

const closePopUp = jest.fn()

const relatedBookId = ""


it("renders for allShelfs", () => {
    
        let books = [
            { id: 10, title: "Livro", authors: "Eu", shelf: "read", imageLinks: { thumbnail: "algo" } } 
        ]
    
        let shelf = "allShelfs"
        expect(shallow(<ShelfsBooks books={books}
            setUpdate={setUpdate}
            objShelfs={allShelfs}
            bookIdUpdate={bookIdUpdate}
            filterRelatedBooks={filterRelatedBooks}
            classPopUp={classPopUp}
            relatedBooks={relatedBooks}
            closePopUp={closePopUp}
            relatedBookId={relatedBookId} />)).toMatchSnapshot();
    });