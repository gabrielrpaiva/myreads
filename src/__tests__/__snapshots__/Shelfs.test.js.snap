// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders for allShelfs 1`] = `
<div
  className="app"
>
  <div
    className="list-books"
  >
    <div
      className="list-books-title"
    >
      <h1>
        MyReads
      </h1>
    </div>
    <div
      className="list-books-content"
    >
      <div>
        <div
          className="bookshelf"
        >
          <h2
            className="bookshelf-title"
          >
            Currently Reading
          </h2>
          <div
            className="bookshelf-books"
          >
            <ListBooks
              allShelfs={
                Object {
                  "currentlyReading": "Currently Reading",
                  "moveto": "Move to..",
                  "none": "None",
                  "read": "Read",
                  "wantToRead": "Want To Read",
                }
              }
              bookIdUpdate=""
              books={
                Array [
                  Object {
                    "authors": "Eu",
                    "id": 10,
                    "imageLinks": Object {
                      "thumbnail": "algo",
                    },
                    "shelf": "read",
                    "title": "Livro",
                  },
                ]
              }
              classPopUp=""
              closePopUp={[Function]}
              filterRelatedBooks={[Function]}
              relatedBookId=""
              relatedBooks={Array []}
              setUpdate={[Function]}
              shelf="currentlyReading"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="list-books-content"
    >
      <div>
        <div
          className="bookshelf"
        >
          <h2
            className="bookshelf-title"
          >
            Want To Read
          </h2>
          <div
            className="bookshelf-books"
          >
            <ListBooks
              allShelfs={
                Object {
                  "currentlyReading": "Currently Reading",
                  "moveto": "Move to..",
                  "none": "None",
                  "read": "Read",
                  "wantToRead": "Want To Read",
                }
              }
              bookIdUpdate=""
              books={
                Array [
                  Object {
                    "authors": "Eu",
                    "id": 10,
                    "imageLinks": Object {
                      "thumbnail": "algo",
                    },
                    "shelf": "read",
                    "title": "Livro",
                  },
                ]
              }
              classPopUp=""
              closePopUp={[Function]}
              filterRelatedBooks={[Function]}
              relatedBookId=""
              relatedBooks={Array []}
              setUpdate={[Function]}
              shelf="wantToRead"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="list-books-content"
    >
      <div>
        <div
          className="bookshelf"
        >
          <h2
            className="bookshelf-title"
          >
            Read
          </h2>
          <div
            className="bookshelf-books"
          >
            <ListBooks
              allShelfs={
                Object {
                  "currentlyReading": "Currently Reading",
                  "moveto": "Move to..",
                  "none": "None",
                  "read": "Read",
                  "wantToRead": "Want To Read",
                }
              }
              bookIdUpdate=""
              books={
                Array [
                  Object {
                    "authors": "Eu",
                    "id": 10,
                    "imageLinks": Object {
                      "thumbnail": "algo",
                    },
                    "shelf": "read",
                    "title": "Livro",
                  },
                ]
              }
              classPopUp=""
              closePopUp={[Function]}
              filterRelatedBooks={[Function]}
              relatedBookId=""
              relatedBooks={Array []}
              setUpdate={[Function]}
              shelf="read"
            />
          </div>
        </div>
      </div>
    </div>
    <Link
      className="open-search"
      replace={false}
      to="/search"
    />
  </div>
</div>
`;
